src/platform/synchronization.rs

```diff
@@ -0,0 +1,292 @@
+use std::{
+    sync::{Condvar, Mutex, MutexGuard},
+    thread::{self, JoinHandle},
+};
+
+use libc::c_void;
+
+use crate::{impl_guarded_transmute, GuardedTransmute};
+
+pub struct ZMutex<'a> {
+    mutex: Mutex<()>,
+    lock: Option<MutexGuard<'a, ()>>,
+}
+
+pub struct ZMutexPtr {
+    data: Option<Box<ZMutex<'static>>>,
+}
+
+/// Mutex
+///
+#[repr(C)]
+#[derive(Clone, Copy)]
+pub struct z_mutex_t(usize);
+
+impl_guarded_transmute!(z_mutex_t, ZMutexPtr);
+impl_guarded_transmute!(ZMutexPtr, z_mutex_t);
+
+// try to use the same error codes as in GNU pthreads
+const EBUSY: i8 = 16;
```

Comment:
This doesn't fit the agreement, that the error value should be always negative. This agreement is used here for example: https://github.com/eclipse-zenoh/zenoh-cpp/blob/main/include/zenohcxx/impl.hxx#L59-L69
Though this agreement is already violated in zenoh-pico by these functions like `z_mutex_init` which directly returns values from pthread. So it seems like that it's necessary to reevaluate the assumption that error value is alvays negative

Reply:
I believe error values should always be negative.
I would consider as a bug the fact that `z_mutex_init` directly returns values from pthread in zenoh-pico.

Reply:
made issue in zenoh-pico
https://github.com/eclipse-zenoh/zenoh-pico/issues/358